test/Object/getOwnPropertyDescriptors.js

```diff
@@ -153,7 +153,29 @@ var tests = [
 
             verifyObjectDescriptors(desc, allTrueSymbol, allFalseSymbol);
         }
-    }
+    },
+    {
+        name:"For any property, if getOwnPropertyDescriptor(property) is undefined, that property should not be present on the result.",
+        body: function() {
+            // Adapted from: https://github.com/ljharb/test262/blob/c2eaa30b08fb1e041b7297e415b6bad8461f50dc/test/built-ins/Object/getOwnPropertyDescriptors/proxy-undefined-descriptor.js
+            var proxyHandler = {
+              getOwnPropertyDescriptor: function () {},
+            };
+
+            var key = "a";
+            var obj = {};
+            obj[key] = "value";
+
+            var proxy = new Proxy(obj, proxyHandler);
+
+            var descriptor = Object.getOwnPropertyDescriptor(proxy, key);
+            assert.areEqual(undefined, descriptor, "Descriptor matches result of [[GetOwnPropertyDescriptor]] trap");
+
+            var result = Object.getOwnPropertyDescriptors(proxy);
+            assert.isFalse(key in result, "key should not be present in result");
```

Comment:
The `in` operator also returns true for properties found in the prototype change.  Here we don't expect the prototype change of a normal plain object to have a property named 'a', but it would be better practice to use `obj.hasOwnProperty(key)` instead